projects/fhi-angular-components/src/lib/fhi-tree-view-selection/fhi-tree-view-selection.component.ts

```diff
@@ -218,14 +214,18 @@ export class FhiTreeViewSelectionComponent implements OnInit, OnChanges {
     return itemsState;
   }
 
-  private createIds(items: Item[], id: number) {
+  private createIds(items: Item[]) {
     items.forEach((item) => {
       if (item.id === undefined) {
-        item.id = id++;
+        item.id = this.getRandomID();
```

Comment:
If it is a problem at an instance level; do you actually need to create a random id for every item?

Is it possible to create a uuid once for each instance and then add an incrementing number for each item instead?

Reply:
I agree, that is a better solution.